code/lib/cli-storybook/src/automigrate/fixes/addon-globals-api.ts

```diff
@@ -310,18 +290,38 @@ function transformStoryFile(
       viewportParams = getObjectProperty(parameters, 'viewport') as ObjectExpression;
       if (viewportParams) {
         const defaultViewport = getObjectProperty(viewportParams, 'defaultViewport');
-        if (defaultViewport && t.isStringLiteral(defaultViewport)) {
+        const defaultOrientation = getObjectProperty(viewportParams, 'defaultOrientation');
+        const disableViewport = getObjectProperty(viewportParams, 'disable');
+
+        // Handle both string literals and member expressions for defaultViewport
+        let viewportValue: t.StringLiteral | t.MemberExpression | null = null;
+        if (defaultViewport) {
+          if (t.isStringLiteral(defaultViewport)) {
+            viewportValue = defaultViewport;
+          } else if (t.isMemberExpression(defaultViewport)) {
+            // Preserve the member expression as-is
+            viewportValue = defaultViewport;
+          }
+        }
+
+        if (viewportValue) {
           // Create globals.viewport
           if (!newGlobals) {
             newGlobals = t.objectExpression([]);
           }
 
+          // Determine isRotated based on defaultOrientation
+          let isRotated = false;
+          if (defaultOrientation && t.isStringLiteral(defaultOrientation)) {
+            isRotated = defaultOrientation.value === 'portrait';
+          }
+
           newGlobals.properties.push(
             t.objectProperty(
               t.identifier('viewport'),
               t.objectExpression([
-                t.objectProperty(t.identifier('value'), t.stringLiteral(defaultViewport.value)),
-                t.objectProperty(t.identifier('isRotated'), t.booleanLiteral(false)),
+                t.objectProperty(t.identifier('value'), viewportValue),
+                t.objectProperty(t.identifier('isRotated'), t.booleanLiteral(isRotated)),
```

Comment:
if there was no defaultOrientation ever defined, does it make sense to add `isRotated` to every story?